go/src/hashrocket/go-websocket-server/handler.go

```diff
@@ -71,16 +72,21 @@ func (h *benchHandler) echo(ws *websocket.Conn, payload interface{}) error {
 func (h *benchHandler) broadcast(ws *websocket.Conn, payload interface{}) error {
 	result := BroadcastResult{Type: "broadcastResult", Payload: payload}
 
-	h.mutex.RLock()
+	msg, err := json.Marshal(&WsMsg{Type: "broadcast", Payload: payload})
+	if err != nil {
+		log.Println("broadcast json marshal err:", err)
```

Comment:
It's consistent with the previous implementation which would only bump the `result.ListenerCount` for non-error conditions and always send a result. Any json marshaling error would occur in `websocket.JSON.Send` before, now it's exposed in the prep to call send.